complaint_search/es_interface.py

```diff
@@ -21,17 +22,155 @@ def get_es():
             timeout=100)
     return _ES_INSTANCE
 
-def search(format="json"):
+def _create_and_append_bool_should_clauses(es_field_name, value_list, 
+    filter_list, with_subitems=False, es_subitem_field_name=None):
+    if value_list:
+        if not with_subitems:
+            term_list = [ {"terms": {es_field_name: [value]}} 
+                for value in value_list ]
+            filter_list.append({"bool": {"should": term_list}})
+        else:
+            item_dict = defaultdict(list)
+            for v in value_list:
+                # -*- coding: utf-8 -*-
+                v_pair = v.split(u'\u2022')
+                print v_pair
+                # No subitem
+                if len(v_pair) == 1:
+                    # This will initialize empty list for item if not in item_dict yet
+                    item_dict[v_pair[0]]
+                elif len(v_pair) == 2:
+                    # put subproduct into list
+                    item_dict[v_pair[0]].append(v_pair[1])
+
+            print "item_dict", item_dict
+            # Go through item_dict to create filters
+            f_list = []
+            for item, subitems in item_dict.iteritems():
+                item_term = {"terms": {es_field_name: [item]}}
+                # Item without any subitems
+                if not subitems:
+                    f_list.append(item_term)
+                else:
+                    subitem_term = {"terms": {es_subitem_field_name: subitems}}
+                    f_list.append({"and": {"filters": [item_term, subitem_term]}})
+
+            filter_list.append({"bool": {"should": f_list}})
+
+# List of possible arguments:
+# - fmt: format to be returned: "json", "csv", "xls", or "xlsx"
+# - field: field you want to search in: "what_happened", "comp_status_archive", "_all"
+# - size: number of complaints to return
+# - frm: from which index to start returning
+# - sort: sort by: "relevance_desc", "relevance_asc", "created_date_desc", "created_date_asc"
+# - search_term: the term to be searched
+# - min_date: return only date including and later than this date i.e. 2017-03-02
+# - max_date: return only date before this date, i.e. 2017-04-12 
+# - company: a list of companies you want ["Bank 1", "Bank 2"]
+# - product: 
+# - issue:
+# - state:
+# - consumer_disputed:
+# - company_response:
+def search(**kwargs):
```

Comment:
Added:
```
zip_code
timely
company_public_response
```

The only fields are either missing from the mapping or I am not sure which mapping they map to.  Will talk to @AdamZarger about those.